README.md
# In the beginning
This ModPack has based on IC2 add-on GregTech unofficial version.
Make changes if there are simple changes that can be made with the machine GregTech recipe, make there alone. And able to create useful machine added in the other Mod is to proceed with it.

# Challenge of this modpack
01. Making the Torcherino and Quarryplus to run them.
02. Making the Ultimate battery.
03. Making the UV solar panel.
04. Making the ME 16384k Storage.
05. Making the Creative only items!

# ModList
Take a look at the download, make sure!

# 初めに
このModPackは、IC2のアドオンであるGregTech非公式版をベースにしている。
簡単な変更点だが、GregTechの機械で作れるレシピが存在するならば、そちらだけで作れるように変更を加える。そして、ほかのModで追加される便利な機械は、段階を進めると作成が出来るようにしていく。

# ModPackの挑戦状
01. TorcherinoとQuarryplusを作成して動かすこと
02. Ultimate batteryを作成すること
03. UVのソーラーパネルを作成すること
04. MEの16384kストレージを作成すること
05. クリエイティブ専用アイテムを作成すること!

# ModList
ぜひダウンロードをして確かめて欲しい!


CHANGELOG.md
# v4.4.1
## ModPack update infomation
- Hot fixes to fix a bug that caused some mods to disappear in the cf version.

## Modpack other infomation
- The 1.12.2 version of GTE based on [GregTech CE: Unofficial](https://www.curseforge.com/minecraft/mc-mods/gregtech-ce-unofficial) is now available!!

## Update mods
- GTTweaker

***

# v4.4.0
## ModPack update infomation
- There was a request for an update, so I updated it.
- The main focus is on QOL, such as updating old mods and switching to later development versions.

## Modpack other infomation
- The 1.12.2 version of GTE based on [GregTech CE: Unofficial](https://www.curseforge.com/minecraft/mc-mods/gregtech-ce-unofficial) is now available!!

## New mods
- JadedLibs

## Move mods
- Extra Cells 2 -> ExtraCells2 - SamLam140330's Fork
- ReAuth -> OAuth
- NotEnoughItems -> NotEnoughItems Unofficial

## Update mods
- AE2 Stuff
- BuildCraft
- Controlling
- EnderCore
- EnderIO
- BdLib
- Cosmetic Armor Reworked
- JourneyMap

## Delete mods
- Fullscreen Windowed
- IC2 Crop-Breeding Plugin
- NotEnoughKeys


serverfiles/settings.sh
# Don't edit these values unless you know what you are doing.
export INSTALL_JAR="forge-1.7.10-10.13.4.1614-1.7.10-installer.jar"
export SERVER_JAR="forge-1.7.10-10.13.4.1614-1.7.10-universal.jar"

# You can edit these values if you wish.
export MIN_RAM="4096M"
export MAX_RAM="8192M"
export Max_MetaspaceSize="512M"
export JAVA_PARAMETERS="-XX:MaxRAMPercentage=95.0 -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Dterminal.jline=false -Dterminal.ansi=true -XX:+DisableExplicitGC -XX:+UseConcMarkSweepGC -XX:+DisableExplicitGC -XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:+UseNUMA -XX:+CMSParallelRemarkEnabled -XX:MaxTenuringThreshold=15 -XX:MaxGCPauseMillis=30 -XX:GCPauseIntervalMillis=150 -XX:+UseAdaptiveGCBoundary -XX:-UseGCOverheadLimit -XX:+UseBiasedLocking -XX:SurvivorRatio=8 -XX:TargetSurvivorRatio=90 -XX:MaxTenuringThreshold=15 -Dfml.ignorePatchDiscrepancies=true -Dfml.ignoreInvalidMinecraftCertificates=true -XX:+UseFastAccessorMethods -XX:+UseCompressedOops -XX:+OptimizeStringConcat -XX:+AggressiveOpts -XX:ReservedCodeCacheSize=2048m -XX:+UseCodeCacheFlushing -XX:SoftRefLRUPolicyMSPerMB=10000 -XX:ParallelGCThreads=10 -Djava.net.preferIPv4Stack=true -Dlog4j.configurationFile=log4j2_112-116.xml"


buildtools/mod-install.py
#!/usr/bin/env python3
# GTE Mod install

## Import library
import json
import os
import shutil
import requests
import time
from concurrent.futures import ThreadPoolExecutor

## Initialize
cf_url = "https://api.curseforge.com"
headers = {
  "Accept": "application/json",
  "x-api-key": os.environ.get("CF_API_TOKEN")
}
tpe = ThreadPoolExecutor(max_workers=20)

## Run main function
def main(key):
    url = "{}/v1/mods/{}/files/{}".format(
        cf_url,
        key["projectID"],
        key["fileID"]
    )
    mod_json = requests.get(url, allow_redirects=True, stream=True, headers=headers)

    if mod_json.status_code == 200:
        mod_json = mod_json.json()
        print("{} - {}".format(mod_json["data"]["fileName"], mod_json["data"]["downloadUrl"]))

        mod_file = requests.get(mod_json["data"]["downloadUrl"], allow_redirects=True, stream=True, headers=headers)
        with open(mod_json["data"]["fileName"], "wb") as f:
            shutil.copyfileobj(mod_file.raw, f)

            # ./overrides/mods/
            if mod_json["data"]["fileName"].endswith(".jar"):
                if key["required"]:
                    shutil.move(mod_json["data"]["fileName"], "./overrides/mods/")
                else:
                    os.rename(mod_json["data"]["fileName"], "./overrides/mods/{}".format(mod_json["data"]["fileName"].replace(".jar", ".jar.disabled")))

            # ./overrides/resourcepacks/
            elif mod_json["data"]["fileName"].endswith(".zip"):
                if key["required"]:
                    shutil.move(mod_json["data"]["fileName"], "./overrides/resourcepacks/")
                else:
                    os.rename(mod_json["data"]["fileName"], "./overrides/resourcepacks/{}".format(mod_json["data"]["fileName"].replace(".zip", ".zip.disabled")))

            print("Downloaded {}".format(mod_json["data"]["fileName"]))
    else:
        tpe.shutdown()
        raise Exception("Status code is not 200 : " + url)


## Import List
manifest_json = json.load(open("./manifest.json"))

## Access CF and save file if request succeeds
for key in manifest_json["files"]:
    tpe.submit(main, key)

## Finalize
tpe.shutdown()
print("finished all download.")


serverfiles/TmuxServer.sh
#!/bin/bash
# Minecraft service that starts the server in a tmux session.
# https://github.com/tier940/minecraft-tmux

JAVA_PATH="/usr/lib/jvm/java-8-openjdk/bin/java"
TMUX_SOCKET="steam"
TMUX_SESSION="gte1"
MC_HOME="/home/minecraft/gte1/"
MC_JAR_FILE="forge-1.7.10-10.13.4.1614-1.7.10-universal.jar"
MC_MIN_RAM="4G"
MC_MAX_RAM="8G"
MC_JVM_PARAMETERS="-XX:MaxMetaspaceSize=512M -XX:MaxRAMPercentage=95.0 -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Dterminal.jline=false -Dterminal.ansi=true -XX:+DisableExplicitGC -XX:+UseConcMarkSweepGC -XX:+DisableExplicitGC -XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:+UseNUMA -XX:+CMSParallelRemarkEnabled -XX:MaxTenuringThreshold=15 -XX:MaxGCPauseMillis=30 -XX:GCPauseIntervalMillis=150 -XX:+UseAdaptiveGCBoundary -XX:-UseGCOverheadLimit -XX:+UseBiasedLocking -XX:SurvivorRatio=8 -XX:TargetSurvivorRatio=90 -XX:MaxTenuringThreshold=15 -Dfml.ignorePatchDiscrepancies=true -Dfml.ignoreInvalidMinecraftCertificates=true -XX:+UseFastAccessorMethods -XX:+UseCompressedOops -XX:+OptimizeStringConcat -XX:+AggressiveOpts -XX:ReservedCodeCacheSize=2048m -XX:+UseCodeCacheFlushing -XX:SoftRefLRUPolicyMSPerMB=10000 -XX:ParallelGCThreads=10 -Djava.net.preferIPv4Stack=true -Dlog4j.configurationFile=log4j2_112-116.xml"

is_server_running() {
    /usr/bin/tmux -L ${TMUX_SOCKET} has-session -t ${TMUX_SESSION} > /dev/null 2>&1
    return ${?}
}

mc_command() {
    cmd="$1"
    /usr/bin/tmux -L ${TMUX_SOCKET} send-keys -t ${TMUX_SESSION}.0 "$cmd" ENTER
    return ${?}
}

start_server() {
    if is_server_running; then
        echo "Server already running"
        return 1
    else
        echo "Starting minecraft server in tmux session"
        /usr/bin/tmux -L ${TMUX_SOCKET} new-session -c ${MC_HOME} -s ${TMUX_SESSION} -d ${JAVA_PATH} -Xms${MC_MIN_RAM} -Xmx${MC_MAX_RAM} ${MC_JVM_PARAMETERS} -jar ${MC_HOME}$MC_JAR_FILE
        return ${?}
    fi
}

stop_server() {
    if ! is_server_running; then
        echo "Server is not running!"
        return 1
    else
        # Warn players
        echo "Warning players"
        mc_command "title @a times 3 14 3"
        for i in {60..1}; do
            mc_command "title @a subtitle {\"text\":\"in ${i} seconds\",\"color\":\"gray\"}"
            mc_command "title @a title {\"text\":\"Shutting down\",\"color\":\"dark_red\"}"
            echo "Server stop in ${i} seconds"
            sleep 1
        done

        # Issue shutdown
        echo "Save all"
        mc_command "save-all"
        sleep 0
        echo "Stopping server"
        mc_command "stop"
        if [[ ${?} -ne 0 ]]; then
            echo "Failed to send stop command to server"
            return 1
        fi

        # Wait for server to stop
        wait=0
        while is_server_running; do
            sleep 1

            wait=$((wait+1))
            if [ ${wait} -gt 90 ]; then
                echo "Could not stop server, timeout"
                return 1
            fi
        done
        return 0
    fi
}

restart_server() {
    if ! is_server_running; then
        echo "Server is not running!"
        return 1
    else
        if stop_server; then
            start_server
            return ${?}
        else
            echo "Failed server restart"
            return 1
        fi
    fi
}

reload_server() {
    /usr/bin/tmux -L ${TMUX_SOCKET} send-keys -t ${TMUX_SESSION}.0 "reload" ENTER
    return ${?}
}

attach_session() {
    if ! is_server_running; then
        echo "Cannot attach to server session, server not running"
        return ${?}
    else
        /usr/bin/tmux -L ${TMUX_SOCKET} attach-session -t ${TMUX_SESSION}
        return 0
    fi
}

case "$1" in
    start)
        start_server
        attach_session
        exit ${?}
        ;;
    stop)
        stop_server
        exit ${?}
        ;;
    restart)
        restart_server
        exit ${?}
        ;;
    reload)
        reload_server
        exit ${?}
        ;;
    attach)
        attach_session
        exit ${?}
        ;;
    *)
        echo "Usage: ${0} {start|stop|restart|reload|attach}"
        exit 2
        ;;
esac
